File: Api/src/company/application/company.go
package application

import (
	"gitlab.clirisgroup.com/internal/account-admin-tool/Api/src/company/domain/entity"
	"gitlab.clirisgroup.com/internal/account-admin-tool/Api/src/company/domain/repository"
)

type CompanyAppIer interface {
	GetAll() ([]entity.Company, error)
	Save(*entity.Company) error
}

type CompanyApp struct {
	db repository.CompanyRepository
}

func NewCompanyApp(db repository.CompanyRepository) *CompanyApp {
	return &CompanyApp{db}
}

func (app *CompanyApp) GetAll() ([]entity.Company, error) {
	return app.db.GetAll()
}

func (app *CompanyApp) Save(company *entity.Company) error {
	return app.db.Save(company)
}

File: frontend/app/company/company.ts
import Company from '../../domain/models/company/company';
import CompanyRepository from '../../domain/ports/company/company';

export async function getAll(companyRepository: CompanyRepository): Promise<Company[]> {
    return await companyRepository.getAll();
}

export async function addCompany(companyRepository: CompanyRepository, name: string): Promise<Company> {
    return await companyRepository.save(name);
}
